src/main/java/jalf/compiler/Cog.java

```diff
@@ -156,5 +158,20 @@ public Cog union(Union union, Cog right) {
         };
         return new BaseCog(union, supplier);
     }
+    /** Default compilation of `intersect`. */
+    public Cog intersect(Intersect intersect, Cog right) {
+        // stream compilation: concat + distinct
+
+        Supplier<Stream<Tuple>> supplier = () ->{
+            Stream<Tuple> leftStream = this.stream();
+            Stream<Tuple> rightStream = right.stream();
+            List<Tuple> leftList = leftStream.collect(Collectors.toList());
+            Stream<Tuple>intersectStream =rightStream
+                    .filter(x -> leftList.contains(x));
+            return intersectStream ;
+
+        };
+        return new BaseCog(intersect, supplier);
+    }
```

Comment:
@anne-france  Complexity is way too high. We should at least use another data structure than a list.